setup.py

```diff
@@ -468,9 +470,41 @@ def _read_requirements(filename: str) -> List[str]:
 package_data = {
     "vllm": ["py.typed", "model_executor/layers/fused_moe/configs/*.json"]
 }
+
 if envs.VLLM_USE_PRECOMPILED:
+    assert _is_cuda(), "VLLM_USE_PRECOMPILED is only supported for CUDA builds"
+
+    import zipfile
+
     ext_modules = []
-    package_data["vllm"].append("*.so")
+
+    wheel_location = os.getenv(
+        "VLLM_PRECOMPILED_WHEEL_LOCATION",
+        "https://vllm-wheels.s3.us-west-2.amazonaws.com/nightly/vllm-1.0.0.dev-cp38-abi3-manylinux1_x86_64.whl"
+    )
+
+    if os.path.exists(wheel_filename := os.path.basename(wheel_location)):
+        print(f"Using existing wheel={wheel_filename}")
+    else:
+        try:
+            subprocess.check_call(
+                f"pip download --no-deps {wheel_location}".split(" "))
+        except subprocess.CalledProcessError as exc:
+            from setuptools.errors import SetupError
+
+            raise SetupError(
+                f"Failed to get vLLM wheel from {wheel_location}") from exc
+
+    with zipfile.ZipFile(wheel_filename) as wheel:
+        for lib in filter(lambda file: file.filename.endswith(".so"),
+                          wheel.filelist):
+            package_name = os.path.dirname(lib.filename).replace("/", ".")
+            if package_name not in package_data:
+                package_data[package_name] = []
+
+            wheel.extract(lib)
+            package_data[package_name].append(lib.filename)
+            print(f"Added {lib.filename} to package_data[\"{package_name}\"]")
```

Comment:
I think we also need to copy some python file from vllm flash attention.

Reply:
Indeed, the previous version was only version because of a previously extracted `vllm_flash_attn` module.

With this patch, building from scratch results in all the required files being present:

```shell
$ unzip -l dist/vllm-0.6.3.post2.dev62+g2acec044e.d20241028.cu126.precompiled-py3-none-any.whl | grep -E flash_attn\|\\\.so
200404640  2024-10-28 15:45   vllm/_C.abi3.so
 82008256  2024-10-28 15:45   vllm/_moe_C.abi3.so
    31950  2024-10-28 09:11   vllm/attention/backends/flash_attn.py
    27715  2024-10-28 09:11   vllm/attention/backends/rocm_flash_attn.py
      347  2024-10-28 15:45   vllm/vllm_flash_attn/__init__.py
    47716  2024-10-28 15:45   vllm/vllm_flash_attn/flash_attn_interface.py
242221240  2024-10-28 15:45   vllm/vllm_flash_attn/vllm_flash_attn_c.abi3.so 
```